src/JSTest.Runner/ObjectModel/Common/TestCase.ts

```diff
@@ -30,5 +36,51 @@ export class TestCase {
             .appendStr(this.Source);
 
         this.Id = hash.getGuid();
+
+        this.AttachmentGuid = null;
+        if (attachmentId) {
+            const attachmentHash = new Md5();
+            attachmentHash.appendStr(attachmentId);
+            this.AttachmentGuid = attachmentHash.getGuid();
+        }
+    }
+
+    /**
+     * Looks for files to upload under the specified folder using test case specific id.
+     * 
+     * @param attachmentsRootFolder Root folder of the attachments.
+     */
+    public getAttachments(attachmentsRootFolder: string): AttachmentSet[] {
+        const attachments = new Array<AttachmentSet>();
+
+        // Lets see any file exists in the attachments folder upload
+        if (attachmentsRootFolder && this.AttachmentGuid) {
+            const attachmentsFolder = path.join(attachmentsRootFolder, this.AttachmentGuid);
+            if (fs.existsSync(attachmentsFolder) && fs.lstatSync(attachmentsFolder).isDirectory()) {
+                let attachmentSet: AttachmentSet = null;
+
+                // Iterate through the files under attachments folder to get the list of attachments
+                fs.readdirSync(attachmentsFolder).forEach(file => {
```

Comment:
try catches for fs operations, also when you catch, log an error in eqttrace and return `[]` i guess?

Reply:
in fact you can remove previous existsSync and lstatSync operations and just depend on readdirSync and it throwing in case it doesn't exist or is not a directory

Reply:
A test may not have any files to upload and in this case the attachment folder doesn't exist and it is fine to return an empty array here.